clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1427,13 +1427,13 @@ OpFoldResult cir::SelectOp::fold(FoldAdaptor adaptor) {
 //===----------------------------------------------------------------------===//
 LogicalResult cir::ShiftOp::verify() {
   mlir::Operation *op = getOperation();
-  mlir::Type resType = getResult().getType();
   const bool isOp0Vec = mlir::isa<cir::VectorType>(op->getOperand(0).getType());
   const bool isOp1Vec = mlir::isa<cir::VectorType>(op->getOperand(1).getType());
   if (isOp0Vec != isOp1Vec)
     return emitOpError() << "input types cannot be one vector and one scalar";
-  if (isOp1Vec && op->getOperand(1).getType() != resType) {
-    return emitOpError() << "shift amount must have the type of the result "
+
+  if (isOp1Vec && !mlir::isa<cir::VectorType>(getResult().getType())) {
```

Comment:
Is the requirement that the operands and the result must have the same size elements enforced somewhere else?

Reply:
I added requirements for element type, width, and vector size because now we don't compare the vector types, I think if we will need this more than one time we can create a function to compare vectors without comparing element type isSigned